analyses/survival-analysis/util/quantile_calc.R

```diff
@@ -2,50 +2,59 @@
-                                    ifelse(meta_indep$tp53_score >= tp53_quantiles[[4]] &  meta_indep$telomerase_score <= telomerase_quantiles[[2]], "tp53_high telomerase_low",
-                                           ifelse(meta_indep$tp53_score <= tp53_quantiles[[2]] &  meta_indep$telomerase_score >= telomerase_quantiles[[4]], "tp53_low telomerase_high",
-                                                  ifelse(meta_indep$tp53_score <= tp53_quantiles[[2]] &  meta_indep$telomerase_score <= telomerase_quantiles[[2]], "tp53_low telomerase_low",
-                                                         ifelse(meta_indep$tp53_score > tp53_quantiles[[2]] & meta_indep$tp53_score < tp53_quantiles[[4]] &  meta_indep$telomerase_score <= telomerase_quantiles[[2]], "tp53_mid telomerase_low",
-                                                                ifelse(meta_indep$tp53_score <= tp53_quantiles[[2]] &  meta_indep$telomerase_score > telomerase_quantiles[[2]] & meta_indep$telomerase_score < telomerase_quantiles[[4]], "tp53_low telomerase_mid",
-                                                                       "tp53 tel mid"))))))
-  
-  table(meta_indep$pheno_strict)
-  
-  ### without mid
-  meta_indep$pheno_extremes <- ifelse(meta_indep$tp53_score >= tp53_quantiles[[4]] & meta_indep$telomerase_score >= telomerase_quantiles[[4]], "tp53_high telomerase_high",
-                                      ifelse(meta_indep$tp53_score >= tp53_quantiles[[4]] &meta_indep$telomerase_score <= telomerase_quantiles[[2]], "tp53_high telomerase_low",
-                                             ifelse(meta_indep$tp53_score <= tp53_quantiles[[3]] & meta_indep$telomerase_score >= telomerase_quantiles[[4]], "tp53_low telomerase_high",
-                                                    ifelse(meta_indep$tp53_score <= tp53_quantiles[[2]] & meta_indep$telomerase_score <= telomerase_quantiles[[2]], "tp53_low telomerase_low", NA))))
+  # extract the quantile infomration 
+  tp53_upper <- tp53_quantiles[[4]]
+  tp53_lower <- tp53_quantiles[[2]]
+  telomerase_upper <- telomerase_quantiles[[4]]
+  telomerase_lower <- telomerase_quantiles[[2]]
+  
+  meta_indep <- meta_indep %>%
+    dplyr::mutate(
+      # use 0.5 as a cutoff for tp53 score and telomerase score
+      tp53_0.5 = dplyr::case_when(
+        tp53_score > 0.5  ~ "tp53_high",
+        tp53_score <= 0.5 ~ "tp53_low", 
+        TRUE ~ NA_character_),
+      tel_0.5 = dplyr::case_when(
+        telomerase_score > 0.5  ~ "telomerase_high",
+        telomerase_score <= 0.5 ~ "telomerase_low",
+        TRUE ~ NA_character_),
+      
+      # for strict ones, 25th and 75th quantile is used
+      tp53_strict = dplyr::case_when(
+        tp53_score >= tp53_upper ~ "tp53_high",
+        tp53_score <= tp53_lower ~ "tp53_low",
+        tp53_score > tp53_lower & tp53_score < tp53_upper ~ "tp53_mid",
```

Comment:
Pretty minor, but for consistency we might want to keep where the "or equal to" in a consistent direction (here and in `tel_strict` below).
```suggestion
        tp53_score >  tp53_upper ~ "tp53_high",
        tp53_score <= tp53_lower ~ "tp53_low",
        tp53_score > tp53_lower & tp53_score <= tp53_upper ~ "tp53_mid",
```

Reply:
Thanks! Updated and reran now.